### PR TITLE
Allow task update event submission

### LINKED ISSUE #97: Allow Task update via Event submission
To allow task Update via event submission add the below action

```
{
  "identifier": "c83deeb4-d580-529e-a6f0-c0c649585014",
  "prefix": 5,
  "title": "Mosquito Collection Close Task",
  "description": "Close Mosquito Task",
  "code": "Mosquito Collection",
  "timingPeriod": {
    "start": "2020-11-17",
    "end": "2020-11-24"
  },
  "reason": "Investigation",
  "goalId": "Mosquito_Collection",
  "subjectCodableConcept": {
    "text": "Task"
  },
  "trigger": [
    {
      "type": "named-event",
      "name": "event-submission",
      "expression": {
        "description": "Trigger when a Mosquito Collection event is submitted",
        "expression": "questionnaire = 'mosquito_collection'"
      }
    }
  ],
  "condition": [
    {
      "kind": "applicability",
      "subjectCodableConcept": {
        "text": "Task"
      },
      "expression": {
        "description": "Mosquito Collection event",
        "expression": "$this.contained.where(Task.code.text.value ='Mosquito Collection').exists()"
      }
    }
  ],
  "dynamicValue": [
    {
      "path": "businessStatus",
      "expression": {
        "expression": "$this.item.where(linkId='task_business_status').answer.value"
      }
    },
    {
      "path": "status",
      "expression": {
        "expression": "'Completed'"
      }
    }
  ],
  "definitionUri": "mosquito_collection_form.json",
  "type": "update"
}
```

- This action will allow close of Mosquito task when a mosquito form is submitted
- Dynamic Value allow for expressions that update the task
- Condition allows only the correct task to be updated
- type is update

The below changes needs to be made

- [x] On Task converter set id from Task identifier
- [x] Amend `org.smartregister.pathevaluator.action.ActionHelper.getSubjectResources(Action, QuestionnaireResponse)` add planIdentifier param and when resource type is Task then return `taskDao.findTasksForEntity(entity, planIdentifier);`
Update the PNC plan to add update actions with the correct actions

- [x] Update `org.smartregister.pathevaluator.task.TaskHelper#updateTask` add `questionnaireResponse` param. On the evaluation, evaluate expression on the questionnaire response which has the task as contained so that one can reference both the task and questionnaire 

- [ ] If an event has If an event has task identifier in the details then return the resources as list with the single list of `org.smartregister.pathevaluator.dao.TaskDao.getTaskByIdentifier(String)` you can use this expression to get task from event`$this.identifier.where(id='taskIdentifier' and definition='details').value`

### PR DESCRIPTION
**Pull request by ekigamba**
Fixes #97

### CODE REVIEW

File: src/main/java/org/smartregister/pathevaluator/action/ActionHelper.java [hunk 1]
```diff
@@ -91,6 +102,29 @@ public static ResourceType getResourceType(Action action) {
...
+						if ("details".equals(definitionUri) && "taskIdentifier".equals(answerValue)) {
+							String taskIdentifier = ((com.ibm.fhir.model.type.String) item.getAnswer().get(0).getValue())
+									.getValue();
+
+							org.smartregister.domain.Task task = taskDao.getTaskByIdentifier(taskIdentifier);
+							if (task != null) {
+								ArrayList<Task> tasks = new ArrayList<>();
+								tasks.add(TaskConverter.convertTasktoFihrResource(task));
+
+								return tasks;
```
Comment (githengi) [inline src/main/java/org/smartregister/pathevaluator/action/ActionHelper.java:121]: Simpify to 
```suggestion
								return Collections.singletonList(TaskConverter.convertTasktoFihrResource(task))
```

File: src/main/java/org/smartregister/pathevaluator/action/ActionHelper.java [hunk 2]
```diff
@@ -91,6 +102,29 @@ public static ResourceType getResourceType(Action action) {
 			case FAMILY:
 			case PERSON:
 				return clientDao.findClientById(entity);
+			case TASK:
+				List<QuestionnaireResponse.Item> items = questionnaireResponse.getItem();
```
Comment (githengi) [inline src/main/java/org/smartregister/pathevaluator/action/ActionHelper.java:106]: check if questionnaireResponse is null

File: src/test/java/org/smartregister/pathevaluator/plan/PlanEvaluatorIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	}
+
+	@Test
+	public void evaluatePlanShouldUpdateTaskBasedOnEventObsValues() {
+		PlanDefinition planDefinition = gson.fromJson(TestData.MOSQUITTO_COLLECTION_CLOSE_PLAN, PlanDefinition.class);
+		Event event = gson.fromJson(TestData.MOSQUITTO_COLLECTION_EVENT, Event.class);
+		QuestionnaireResponse eventQuestionnaire = EventConverter.convertEventToEncounterResource(event);
+
+		Task task = gson.fromJson(TestData.MOSQUITTO_COLLECTION_TASK, Task.class);
+		Mockito.doReturn(task).when(taskDao).getTaskByIdentifier(Mockito.anyString());
```
Comment (githengi) [inline src/test/java/org/smartregister/pathevaluator/plan/PlanEvaluatorIntegrationTest.java:66]: Can you verify that this call way made

File: src/main/java/org/smartregister/pathevaluator/action/ActionHelper.java (hunk 2 above)
Comment (ekigamba) [reply src/main/java/org/smartregister/pathevaluator/action/ActionHelper.java:106]: QuestionnaireResponse cannot be null here. There's a check before this call and an operation on L96

**Review by githengi — DISMISSED** [review]

**Review by githengi — APPROVED** [review]

**Pull request merged**